### PR TITLE
Improve compatibility for Powwow and JMC scripts

### PR DESCRIPTION
**Pull request by nschimme**
This PR significantly improves the fidelity of the script converter when dealing with legacy MUD scripts from Powwow and JMC. 

Key improvements:
- **Systematic Abbreviations**: Commands like `#al`, `#ac`, `#unac`, etc., are now handled via a unified mapping rather than ad-hoc hacks.
- **Context-Aware Parameter Mapping**: JMC triggers now correctly shift `%0-%9` to `%1-%10` for TinTin++, while JMC aliases correctly map `%0` to the TinTin++ "all arguments" variable.
- **Robust Expression Parsing**: Powwow expressions now protect quoted strings, ensuring characters like `%` and `+` are not misinterpreted as operators when they appear in text.
- **Expanded Command Support**: Added or improved handling for `#temp`, `#tick` synchronization, `#log` modes, and secure communication commands (`#daa`, `#hide`, `#whisper`).
- **Real-World Verification**: New tests were added using complex scripts from ElvenRunes (e.g., Sìn's teleport system, Axel's XP counter) and Yllemo to ensure the converter respects the original author's intent.
- **Repo Cleanup**: Removed several temporary test scripts and build artifacts from the root directory.

---
*PR created automatically by Jules for task [18083351503348759660](https://jules.google.com/task/18083351503348759660) started by @nschimme*

## Summary by Sourcery

Improve TinTin++ script conversion fidelity for legacy Powwow and JMC scripts, especially around expressions, parameter handling, and command coverage.

New Features:
- Add support for Powwow #temp as a oneshot line action.
- Support JMC secure send commands (#daa, #hide, #whisper) by sending, gagging the line, and annotating with NOP comments.
- Handle JMC #tick display, extended #log modes, and map empty #showme/#output/#echo to #LINE PRINT.
- Initialize Powwow reserved helper functions and aliases via a centralized list of function names and implementations.

Bug Fixes:
- Protect Powwow quoted strings during expression parsing so operators like % and + inside literals are not misinterpreted.
- Preserve numeric addition in Powwow expressions by only collapsing + when string concatenation is intended.
- Correct handling of delayed and standard Powwow/JMC parameters, including special treatment for 0 and negative indices and shifting JMC trigger parameters.
- Fix classification of JMC comment/nop-like commands so they map to #NOP correctly and do not break command splitting.
- Ensure command splitting logic in the TinTin generator recognizes all supported converter commands to avoid accidental line merges.

Enhancements:
- Refine Powwow expression operator handling, including unary %, word/char accessors, and search helpers, using placeholders and reserved function mapping.
- Centralize Powwow reserved function names in a shared utility and generate their TinTin++ implementations programmatically.
- Improve JMC handler behavior for aliases, actions, highlights, substitutes, and gags by marking trigger contexts and adjusting parameter mapping.
- Unify and expand short-form/alias command support for both Powwow and JMC handlers (e.g., al/ac/setvar/var/sub/unac/etc.) via alias maps.
- Simplify converted numbered variable naming by switching from indexed notation to flat suffixes (e.g., powwow_at_7, jmc_dollar_7).

Tests:
- Add unit tests covering Powwow string protection, numeric math, and updated numbered variable naming, including zero and negative indices.
- Extend JMC tests for parameter shifting in triggers vs aliases, #log behavior, empty display commands, #tick display, and secure send commands.
- Add real-world regression tests for ElvenRunes and Yllemo scripts, including sunrise tracking, direction markers, and secure send payloads to validate converter intent.

Chores:
- Remove legacy conversion helper scripts and temporary test artifacts from the repository root.

### CODE REVIEW

**Comment by nschimme** [comment]
@sourcery-ai review

**Comment by nschimme** [comment]
@sourcery-ai review

File: src/converter/generator.js [hunk 1]
```diff
@@ -23,16 +25,27 @@ export class TinTinGenerator {
...
+            const implementations = {
+                'timer': '#FORMAT {now} {%T};#MATH {result} {($now - $powwow_start_time) * 1000}',
+                'rand': '#MATH {result} {1d%1 - 1}',
+                'char_length': '#FORMAT {result} {%x} {%1}',
+                'search': '#FORMAT {result} {%p} {%2} {%1}',
+                'word_count': '#LIST {p_words} {EXPLODE} { }; #LIST {p_words} {FILTER} {{.+}}; #RETURN &p_words[]',
+                'word': '#LIST {p_words} {EXPLODE} { }; #LIST {p_words} {FILTER} {{.+}}; #RETURN $p_words[%2]',
+                'word_slice_to_end': '#LIST {p_words} {EXPLODE} { }; #LIST {p_words} {FILTER} {{.+}}; #LIST {p_slice} {CREATE} {$p_words[%2..-1]}; #LIST {p_slice} {COLLAPSE} { }; #RETURN $p_slice',
+                'word_slice_from_start': '#LIST {p_words} {EXPLODE} { }; #LIST {p_words} {FILTER} {{.+}}; #LIST {p_slice} {CREATE} {$p_words[1..%2]}; #LIST {p_slice} {COLLAPSE} { }; #RETURN $p_slice',
+                'first_char_ascii': '#FORMAT {result} {%a} {%1}',
```
Comment (SourceryAI) [inline src/converter/generator.js:38]: **issue (bug_risk):** `POWWOW_RESERVED_FUNCS` is used here but never defined in this module, causing a runtime/reference error.

Since `POWWOW_RESERVED_FUNCS` currently lives in `utils.js` and isn’t imported or passed into this module, this loop will fail at generation time. Please either import it here or accept it as a parameter (e.g., via the converter) so this module remains self‑contained and avoids the `ReferenceError`.

File: tests/elvenrunes.test.js [hunk 2]
```diff
@@ -41,4 +41,47 @@ describe('Real-world scripts from elvenrunes.com', () => {
...
+
+  it('converts Torq\'s Sunrise Script for JMC', () => {
+    const input = [
+      '#action {^The day has begun.} {#var sunrise 1}',
+      '#action {^The sun rises in the east.} {#var sunrise 1}',
+      '#alias {check_sunrise} {#if {$sunrise == 1} {say It is daytime} {say It is nighttime}}'
+    ].join('\n');
+    const output = jmcConverter.convert(input);
+    expect(output).toContain('#ACTION {^The day has begun.} {#VARIABLE {j_sunrise} {1}}');
+    expect(output).toContain('#ALIAS {check_sunrise} {#IF {$j_sunrise == 1} {say It is daytime} {#ELSE} {say It is nighttime}}');
```
Comment (SourceryAI) [inline tests/elvenrunes.test.js:53]: **suggestion (testing):** Add coverage for JMC #tick and tick* behavior mentioned in the PR description.

This test suite should also cover the new JMC tick behavior (`#tick`, `#ticksize`, `#tickon`, `#tickset`, `#tickoff`). Please add tests that assert `#tick` is rendered as remaining time and that the tick size/on/off commands emit the expected TinTin++ `#TICKER`/`#UNTICKER` sequences and variables, either here or in a dedicated JMC tick describe block, to lock in the new behavior and prevent regressions.

Suggested implementation:

```javascript
    const output = jmcConverter.convert(input);
    expect(output).toContain('#ALIAS {%0 hits, %1 mana, and %2 moves.} {#ALIAS {report} {emote has %0 hps, %1 mana, and %2 moves.}}');
  });


  it('converts Torq\'s Sunrise Script for JMC', () => {
    const input = [
      '#action {^The day has begun.} {#var sunrise 1}',
      '#action {^The sun rises in the east.} {#var sunrise 1}',
      '#alias {check_sunrise} {#if {$sunrise == 1} {say It is daytime} {say It is nighttime}}'
    ].join('\n');
    const output = jmcConverter.convert(input);
    expect(output).toContain('#ACTION {^The day has begun.} {#VARIABLE {j_sunrise} {1}}');
    expect(output).toContain('#ALIAS {check_sunrise} {#IF {$j_sunrise == 1} {say It is daytime} {#ELSE} {say It is nighttime}}');
  });

  describe('JMC tick commands', () => {
    it('converts #tick to a TinTin++ ticker that tracks remaining time', () => {
      const input = [
        '#tick 60 {say Tick!}',
      ].join('\n');

      const output = jmcConverter.convert(input);

      // JMC #tick should become a TinTin++ ticker with a fixed interval and a remaining-time variable.
      // The exact ticker name/variable names should match the converter implementation.
      expect(output).toContain('#TICKER {jmc_tick}');
      expect(output).toContain('{60}');
      expect(output).toContain('say Tick!');
      // Remaining time should be represented via a variable so that scripts can echo it.
      // Adjust the variable name to the one actually used by the implementation.
      expect(output).toMatch(/#VARIABLE \{jmc_tick_remaining\} \{\d+\}/);
    });

    it('converts #ticksize to update the TinTin++ ticker interval and size variable', () => {
      const input = [
        '#tick 30 {say Tick!}',
        '#ticksize 90',
      ].join('\n');

      const output = jmcConverter.convert(input);

      // Initial tick definition
      expect(output).toContain('#TICKER {jmc_tick} {30} {say Tick!}');

      // #ticksize should emit an updated ticker interval and a size variable for JMC compatibility.
      // Adjust expected ticker name / variable names to match the implementation.
      expect(output).toContain('#VARIABLE {jmc_tick_size} {90}');
      expect(output).toContain('#TICKER {jmc_tick} {90}');
    });

    it('converts #tickon and #tickoff to the expected #TICKER/#UNTICKER sequences', () => {
      const input = [
        '#tick 60 {say Tick!}',
        '#tickoff',
        '#tickon',
      ].join('\n');

      const output = jmcConverter.convert(input);

      // Base ticker definition
      expect(output).toContain('#TICKER {jmc_tick} {60} {say Tick!}');

      // Turning ticking off should map to an UNTICKER for the same ticker name.
      expect(output).toContain('#UNTICKER {jmc_tick}');

      // Turning ticking on again should recreate or restart the same ticker.
      expect(output).toMatch(/#TICKER \{jmc_tick\} \{60\} \{.*Tick!.*\}/);
    });

    it('converts #tickset to adjust remaining time for the JMC tick', () => {
      const input = [
        '#tick 60 {say Tick!}',
        '#tickset 15',
      ].join('\n');

      const output = jmcConverter.convert(input);

      // #tickset should change only the remaining time while keeping the base tick size.
      // Adjust variable / command names to match the implementation.
      expect(output).toContain('#TICKER {jmc_tick} {60} {say Tick!}');
      expect(output).toContain('#VARIABLE {jmc_tick_remaining} {15}');
    });
  });

```

The expectations for ticker names (`jmc_tick`), the remaining-time variable (`jmc_tick_remaining`), and the tick-size variable (`jmc_tick_size`) are inferred from the PR description and may not exactly match the existing converter implementation.

To fully integrate these tests you should:

1. Align ticker and variable names:
   - Update `jmc_tick`, `jmc_tick_remaining`, and `jmc_tick_size` in the expectations to the exact identifiers used by the JMC-to-TinTin++ converter (e.g. `j_tick`, `j_tick_remain`, etc.).
2. Confirm argument ordering for `#TICKER`:
   - If your converter emits `#TICKER {name} {command} {interval}` or another order, adjust the `toContain` / `toMatch` expectations accordingly.
3. Confirm how `#tickset` is implemented:
   - If `#tickset` is implemented by re-emitting a `#TICKER` with a different interval rather than a separate remaining-time variable, change the corresponding assertions to check the updated `#TICKER` line instead of `#VARIABLE {jmc_tick_remaining} {...}`.
4. If the converter emits additional helper aliases or variables as part of the tick emulation, you may wish to extend these tests with extra `expect(...).toContain(...)` assertions to lock in that behavior as well.

File: tests/elvenrunes.test.js [hunk 3]
```diff
@@ -41,4 +41,47 @@ describe('Real-world scripts from elvenrunes.com', () => {
...
+  it('converts Jahara\'s Direction Marker for PowTTY', () => {
+    const input = [
+      '#al mark=#mark $1=yellow',
+      '#al unmark=#reset marks',
+      '#al markall=#mark $0=bold red',
+      '#al multi=#mark $1 $2=GREEN'
+    ].join('\n');
+    const output = pwConverter.convert(input);
+    expect(output).toContain('#ALIAS {mark} {#HIGHLIGHT {%1} {yellow}}');
+    expect(output).toContain('#ALIAS {unmark} {#KILL HIGHLIGHTS {*}*}');
```
Comment (SourceryAI) [inline tests/elvenrunes.test.js:65]: **suggestion (testing):** Add tests for Powwow #temp and #exe variants introduced in this PR.

Since this test already validates real-world Powwow alias/mark behavior, it’s a good place (or `converter.test.js`) to cover the new `#temp`/`#exe` support. Please add cases for:
- `#temp` with a braced pattern and command, asserting it becomes a one-shot `#LINE ONESHOT #ACTION` with the expected pattern/body.
- A non-matching or malformed `#temp`, asserting the `#NOP UNCONVERTED TEMP` fallback.
- `#exe !<cmd>`, `#exe <file>`, and `#exe (<expr>)`, ensuring each variant generates the correct script/read/expression-eval output.
These paths are core to the new feature and currently untested.

Suggested implementation:

```javascript
  it('converts Jahara\'s Direction Marker for PowTTY', () => {
    const input = [
      '#al mark=#mark $1=yellow',
      '#al unmark=#reset marks',
      '#al markall=#mark $0=bold red',
      '#al multi=#mark $1 $2=GREEN'
    ].join('\n');
    const output = pwConverter.convert(input);
    expect(output).toContain('#ALIAS {mark} {#HIGHLIGHT {%1} {yellow}}');
    expect(output).toContain('#ALIAS {unmark} {#KILL HIGHLIGHTS {*}*}');
    expect(output).toContain('#ALIAS {markall} {#HIGHLIGHT {%0} {bold red}}');
    expect(output).toContain('#ALIAS {multi} {#HIGHLIGHT {%1 %2} {GREEN}}');
  });

  it('converts Powwow #temp to a one-shot action for PowTTY', () => {
    const input = [
      '#temp {^You are hungry.} {eat bread}'
    ].join('\n');
    const output = pwConverter.convert(input);

    // Should become a one-shot action with the expected pattern and body
    expect(output).toContain('#LINE ONESHOT #ACTION {^You are hungry.} {eat bread}');
  });

  it('falls back to #NOP for malformed Powwow #temp in PowTTY', () => {
    const input = [
      // Missing braces / malformed format
      '#temp ^You are hungry. eat bread'
    ].join('\n');
    const output = pwConverter.convert(input);

    // Malformed variants should not be converted and must emit the UNCONVERTED TEMP fallback
    expect(output).toContain('#NOP');
    expect(output).toContain('UNCONVERTED TEMP');
  });

  it('converts Powwow #exe variants for PowTTY', () => {
    const input = [
      '#exe !echo hello',      // shell/script style execution
      '#exe script.pow',       // read/execute file
      '#exe (1 + 2)'           // evaluate expression
    ].join('\n');
    const output = pwConverter.convert(input);

    // !<cmd> -> script/command execution
    expect(output).toContain('#SCRIPT {echo hello}');

    // <file> -> file read/exec
    expect(output).toContain('#READ {script.pow}');

    // (<expr>) -> expression evaluation
    expect(output).toContain('#EVAL {(1 + 2)}');
  });

```

The exact PowTTY output commands for `#temp` and `#exe` may differ from the placeholders used here (`#LINE ONESHOT #ACTION`, `#SCRIPT`, `#READ`, `#EVAL`, and the `#NOP UNCONVERTED TEMP` message). Please align the `expect(...).toContain(...)` strings with the actual output formats implemented in the converter:
1. Confirm the precise one-shot action syntax generated for `#temp` (e.g. whether `#LINE ONESHOT` and spacing/bracing match) and update that assertion accordingly.
2. Verify the exact fallback string for malformed `#temp` (e.g. `'#NOP UNCONVERTED TEMP'`, or including the original line) and adjust the expectations to match the real message.
3. Check how `#exe !<cmd>`, `#exe <file>`, and `#exe (<expr>)` are mapped (e.g. to `#SCRIPT`, `#READ`, `#EVAL` or other PowTTY-specific commands) and update the expected substrings to those real commands and argument formats.
4. If the converter emits multiple lines per input line, you may want to break the `#exe` test into three separate `it(...)` blocks with one input each, and assert more specific matches (such as `output.split('\n')` containing the exact converted line).

**Review by SourceryAI — COMMENTED** [review]
Hey - I've found 3 issues, and left some high level feedback:

- In `src/converter/generator.js`, you're now iterating over `POWWOW_RESERVED_FUNCS` but that symbol isn't imported in this file, so you should add the corresponding import (as you did in `converter.js`) to avoid a runtime reference error.

<details>
<summary>Prompt for AI Agents</summary>

~~~markdown
Please address the comments from this code review:

## Overall Comments
- In `src/converter/generator.js`, you're now iterating over `POWWOW_RESERVED_FUNCS` but that symbol isn't imported in this file, so you should add the corresponding import (as you did in `converter.js`) to avoid a runtime reference error.

## Individual Comments

### Comment 1
<location path="src/converter/generator.js" line_range="29-38" />
<code_context>
+            const implementations = {
</code_context>
<issue_to_address>
**issue (bug_risk):** `POWWOW_RESERVED_FUNCS` is used here but never defined in this module, causing a runtime/reference error.

Since `POWWOW_RESERVED_FUNCS` currently lives in `utils.js` and isn’t imported or passed into this module, this loop will fail at generation time. Please either import it here or accept it as a parameter (e.g., via the converter) so this module remains self‑contained and avoids the `ReferenceError`.
</issue_to_address>

### Comment 2
<location path="tests/elvenrunes.test.js" line_range="45-53" />
<code_context>
     expect(output).toContain('#ALIAS {%0 hits, %1 mana, and %2 moves.} {#ALIAS {report} {emote has %0 hps, %1 mana, and %2 moves.}}');
   });
+
+  it('converts Torq\'s Sunrise Script for JMC', () => {
+    const input = [
+      '#action {^The day has begun.} {#var sunrise 1}',
+      '#action {^The sun rises in the east.} {#var sunrise 1}',
+      '#alias {check_sunrise} {#if {$sunrise == 1} {say It is daytime} {say It is nighttime}}'
+    ].join('\n');
+    const output = jmcConverter.convert(input);
+    expect(output).toContain('#ACTION {^The day has begun.} {#VARIABLE {j_sunrise} {1}}');
+    expect(output).toContain('#ALIAS {check_sunrise} {#IF {$j_sunrise == 1} {say It is daytime} {#ELSE} {say It is nighttime}}');
+  });
+
</code_context>
<issue_to_address>
**suggestion (testing):** Add coverage for JMC #tick and tick* behavior mentioned in the PR description.

This test suite should also cover the new JMC tick behavior (`#tick`, `#ticksize`, `#tickon`, `#tickset`, `#tickoff`). Please add tests that assert `#tick` is rendered as remaining time and that the tick size/on/off commands emit the expected TinTin++ `#TICKER`/`#UNTICKER` sequences and variables, either here or in a dedicated JMC tick describe block, to lock in the new behavior and prevent regressions.

Suggested implementation:

```javascript
    const output = jmcConverter.convert(input);
    expect(output).toContain('#ALIAS {%0 hits, %1 mana, and %2 moves.} {#ALIAS {report} {emote has %0 hps, %1 mana, and %2 moves.}}');
  });


  it('converts Torq\'s Sunrise Script for JMC', () => {
    const input = [
      '#action {^The day has begun.} {#var sunrise 1}',
      '#action {^The sun rises in the east.} {#var sunrise 1}',
      '#alias {check_sunrise} {#if {$sunrise == 1} {say It is daytime} {say It is nighttime}}'
    ].join('\n');
    const output = jmcConverter.convert(input);
    expect(output).toContain('#ACTION {^The day has begun.} {#VARIABLE {j_sunrise} {1}}');
    expect(output).toContain('#ALIAS {check_sunrise} {#IF {$j_sunrise == 1} {say It is daytime} {#ELSE} {say It is nighttime}}');
  });

  describe('JMC tick commands', () => {
    it('converts #tick to a TinTin++ ticker that tracks remaining time', () => {
      const input = [
        '#tick 60 {say Tick!}',
      ].join('\n');

      const output = jmcConverter.convert(input);

      // JMC #tick should become a TinTin++ ticker with a fixed interval and a remaining-time variable.
      // The exact ticker name/variable names should match the converter implementation.
      expect(output).toContain('#TICKER {jmc_tick}');
      expect(output).toContain('{60}');
      expect(output).toContain('say Tick!');
      // Remaining time should be represented via a variable so that scripts can echo it.
      // Adjust the variable name to the one actually used by the implementation.
      expect(output).toMatch(/#VARIABLE \{jmc_tick_remaining\} \{\d+\}/);
    });

    it('converts #ticksize to update the TinTin++ ticker interval and size variable', () => {
      const input = [
        '#tick 30 {say Tick!}',
        '#ticksize 90',
      ].join('\n');

      const output = jmcConverter.convert(input);

      // Initial tick definition
      expect(output).toContain('#TICKER {jmc_tick} {30} {say Tick!}');

      // #ticksize should emit an updated ticker interval and a size variable for JMC compatibility.
      // Adjust expected ticker name / variable names to match the implementation.
      expect(output).toContain('#VARIABLE {jmc_tick_size} {90}');
      expect(output).toContain('#TICKER {jmc_tick} {90}');
    });

    it('converts #tickon and #tickoff to the expected #TICKER/#UNTICKER sequences', () => {
      const input = [
        '#tick 60 {say Tick!}',
        '#tickoff',
        '#tickon',
      ].join('\n');

      const output = jmcConverter.convert(input);

      // Base ticker definition
      expect(output).toContain('#TICKER {jmc_tick} {60} {say Tick!}');

      // Turning ticking off should map to an UNTICKER for the same ticker name.
      expect(output).toContain('#UNTICKER {jmc_tick}');

      // Turning ticking on again should recreate or restart the same ticker.
      expect(output).toMatch(/#TICKER \{jmc_tick\} \{60\} \{.*Tick!.*\}/);
    });

    it('converts #tickset to adjust remaining time for the JMC tick', () => {
      const input = [
        '#tick 60 {say Tick!}',
        '#tickset 15',
      ].join('\n');

      const output = jmcConverter.convert(input);

      // #tickset should change only the remaining time while keeping the base tick size.
      // Adjust variable / command names to match the implementation.
      expect(output).toContain('#TICKER {jmc_tick} {60} {say Tick!}');
      expect(output).toContain('#VARIABLE {jmc_tick_remaining} {15}');
    });
  });

```

The expectations for ticker names (`jmc_tick`), the remaining-time variable (`jmc_tick_remaining`), and the tick-size variable (`jmc_tick_size`) are inferred from the PR description and may not exactly match the existing converter implementation.

To fully integrate these tests you should:

1. Align ticker and variable names:
   - Update `jmc_tick`, `jmc_tick_remaining`, and `jmc_tick_size` in the expectations to the exact identifiers used by the JMC-to-TinTin++ converter (e.g. `j_tick`, `j_tick_remain`, etc.).
2. Confirm argument ordering for `#TICKER`:
   - If your converter emits `#TICKER {name} {command} {interval}` or another order, adjust the `toContain` / `toMatch` expectations accordingly.
3. Confirm how `#tickset` is implemented:
   - If `#tickset` is implemented by re-emitting a `#TICKER` with a different interval rather than a separate remaining-time variable, change the corresponding assertions to check the updated `#TICKER` line instead of `#VARIABLE {jmc_tick_remaining} {...}`.
4. If the converter emits additional helper aliases or variables as part of the tick emulation, you may wish to extend these tests with extra `expect(...).toContain(...)` assertions to lock in that behavior as well.
</issue_to_address>

### Comment 3
<location path="tests/elvenrunes.test.js" line_range="56-65" />
<code_context>
+    expect(output).toContain('#ALIAS {check_sunrise} {#IF {$j_sunrise == 1} {say It is daytime} {#ELSE} {say It is nighttime}}');
+  });
+
+  it('converts Jahara\'s Direction Marker for PowTTY', () => {
+    const input = [
+      '#al mark=#mark $1=yellow',
+      '#al unmark=#reset marks',
+      '#al markall=#mark $0=bold red',
+      '#al multi=#mark $1 $2=GREEN'
+    ].join('\n');
+    const output = pwConverter.convert(input);
+    expect(output).toContain('#ALIAS {mark} {#HIGHLIGHT {%1} {yellow}}');
+    expect(output).toContain('#ALIAS {unmark} {#KILL HIGHLIGHTS {*}*}');
+    expect(output).toContain('#ALIAS {markall} {#HIGHLIGHT {%0} {bold red}}');
+    expect(output).toContain('#ALIAS {multi} {#HIGHLIGHT {%1 %2} {GREEN}}');
+  });
+
</code_context>
<issue_to_address>
**suggestion (testing):** Add tests for Powwow #temp and #exe variants introduced in this PR.

Since this test already validates real-world Powwow alias/mark behavior, it’s a good place (or `converter.test.js`) to cover the new `#temp`/`#exe` support. Please add cases for:
- `#temp` with a braced pattern and command, asserting it becomes a one-shot `#LINE ONESHOT #ACTION` with the expected pattern/body.
- A non-matching or malformed `#temp`, asserting the `#NOP UNCONVERTED TEMP` fallback.
- `#exe !<cmd>`, `#exe <file>`, and `#exe (<expr>)`, ensuring each variant generates the correct script/read/expression-eval output.
These paths are core to the new feature and currently untested.

Suggested implementation:

```javascript
  it('converts Jahara\'s Direction Marker for PowTTY', () => {
    const input = [
      '#al mark=#mark $1=yellow',
      '#al unmark=#reset marks',
      '#al markall=#mark $0=bold red',
      '#al multi=#mark $1 $2=GREEN'
    ].join('\n');
    const output = pwConverter.convert(input);
    expect(output).toContain('#ALIAS {mark} {#HIGHLIGHT {%1} {yellow}}');
    expect(output).toContain('#ALIAS {unmark} {#KILL HIGHLIGHTS {*}*}');
    expect(output).toContain('#ALIAS {markall} {#HIGHLIGHT {%0} {bold red}}');
    expect(output).toContain('#ALIAS {multi} {#HIGHLIGHT {%1 %2} {GREEN}}');
  });

  it('converts Powwow #temp to a one-shot action for PowTTY', () => {
    const input = [
      '#temp {^You are hungry.} {eat bread}'
    ].join('\n');
    const output = pwConverter.convert(input);

    // Should become a one-shot action with the expected pattern and body
    expect(output).toContain('#LINE ONESHOT #ACTION {^You are hungry.} {eat bread}');
  });

  it('falls back to #NOP for malformed Powwow #temp in PowTTY', () => {
    const input = [
      // Missing braces / malformed format
      '#temp ^You are hungry. eat bread'
    ].join('\n');
    const output = pwConverter.convert(input);

    // Malformed variants should not be converted and must emit the UNCONVERTED TEMP fallback
    expect(output).toContain('#NOP');
    expect(output).toContain('UNCONVERTED TEMP');
  });

  it('converts Powwow #exe variants for PowTTY', () => {
    const input = [
      '#exe !echo hello',      // shell/script style execution
      '#exe script.pow',       // read/execute file
      '#exe (1 + 2)'           // evaluate expression
    ].join('\n');
    const output = pwConverter.convert(input);

    // !<cmd> -> script/command execution
    expect(output).toContain('#SCRIPT {echo hello}');

    // <file> -> file read/exec
    expect(output).toContain('#READ {script.pow}');

    // (<expr>) -> expression evaluation
    expect(output).toContain('#EVAL {(1 + 2)}');
  });

```

The exact PowTTY output commands for `#temp` and `#exe` may differ from the placeholders used here (`#LINE ONESHOT #ACTION`, `#SCRIPT`, `#READ`, `#EVAL`, and the `#NOP UNCONVERTED TEMP` message). Please align the `expect(...).toContain(...)` strings with the actual output formats implemented in the converter:
1. Confirm the precise one-shot action syntax generated for `#temp` (e.g. whether `#LINE ONESHOT` and spacing/bracing match) and update that assertion accordingly.
2. Verify the exact fallback string for malformed `#temp` (e.g. `'#NOP UNCONVERTED TEMP'`, or including the original line) and adjust the expectations to match the real message.
3. Check how `#exe !<cmd>`, `#exe <file>`, and `#exe (<expr>)` are mapped (e.g. to `#SCRIPT`, `#READ`, `#EVAL` or other PowTTY-specific commands) and update the expected substrings to those real commands and argument formats.
4. If the converter emits multiple lines per input line, you may want to break the `#exe` test into three separate `it(...)` blocks with one input each, and assert more specific matches (such as `output.split('\n')` containing the exact converted line).
</issue_to_address>
~~~

</details>

***

Hi @nschimme! 👋

Thanks for trying out Sourcery by commenting with `@sourcery-ai review`! 🚀

Install the [sourcery-ai bot](https://github.com/apps/sourcery-ai/installations/new) to get automatic code reviews on every pull request ✨

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment and I'll use the feedback to improve your reviews.
</sub>

**Comment by nschimme** [comment]
@SourceryAI review
